### PR TITLE
add a format 'len(Data): int' to count elements in a slice

### PR DESCRIPTION
**Pull request by craftgear**
Add a new format to count elements in a response slice.

This would be useful when you expect an exact amount of data to retrieve.

Usage:

```
## Get /foo/total
Get 10 items
=== 
len(Data): 10
```

### CODE REVIEW

**Comment by matryer** [comment]
Great stuff.

Since we're using JavaScript notation, would it make sense to do Data.items.length?

**Comment by craftgear** [comment]
That makes sense.

Then how about making it as ``Data.length``?
When you get response as an array ``response.length`` is intuitive IMHO, moreover it's shorter.

**Comment by craftgear** [comment]
Wait, ``length`` should work with properties and strings.

Let me think about this again.

**Comment by matryer** [comment]
Is this ready to go now?

**Comment by craftgear** [comment]
Not yet.
Give me a little more time.

**Comment by davidbgk** [comment]
Maybe `Data.length(): 6` would avoid the collision with a `length` attribute?

**Comment by matryer** [comment]
Maybe it’s not a collision… you might well want to assert the length of a string?

**Comment by craftgear** [comment]
@matryer Yes, I try to check ``String.length``.
Do you think it's unnecessary?

File: runner/run.go [hunk 1]
```diff
@@ -270,7 +271,23 @@ func (r *Runner) assertData(data interface{}, errData error, key string, expecte
 		r.log(key, fmt.Sprintf("expected %s: %s  actual: no data", expected.Type(), expected))
 		return false
 	}
+
 	actual, ok := m.GetOK(map[string]interface{}{"Data": data}, key)
+
+	if strings.HasSuffix(key, ".length") {
+		key = strings.Replace(key, ".length", "", 1)
```
Comment (matryer) [inline runner/run.go]: TrimSuffix here?

File: runner/run.go [hunk 2]
```diff
@@ -270,7 +271,23 @@ func (r *Runner) assertData(data interface{}, errData error, key string, expecte
...
+		actual, ok = m.GetOK(map[string]interface{}{"Data": data}, key)
+
+		if ok {
+			if slice, _ok := actual.([]interface{}); _ok {
+				actual = float64(len(slice))
+			}
+			if str, _ok := actual.(string); _ok {
+				actual = float64(utf8.RuneCountInString(str))
```
Comment (matryer) [inline runner/run.go]: Why `RuneCountInString` and not just `len`?

File: runner/run.go [hunk 3]
```diff
@@ -198,7 +199,7 @@ func (r *Runner) runRequest(group *parse.Group, req *parse.Request) {
 	if len(req.ExpectedDetails) > 0 {
 		for _, line := range req.ExpectedDetails {
 			detail := line.Detail()
-			if strings.HasPrefix(detail.Key, "Data") {
+			if strings.HasPrefix(detail.Key, "Data") || strings.HasSuffix(detail.Key, ".length") {
```
Comment (matryer) [inline runner/run.go]: Doesn't it _always_ need `Data` prefix?

**Comment by matryer** [comment]
I'd love to get this done and into the next pre-release if possible. @craftgear can you have a look at my comments please?

**Pull request closed without merging**